### PR TITLE
Refactor storage tests - part 3

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/109831

This further refactors the package to build a set of generic storage tests.

```release-note
NONE
```

/sig api-machinery
/kind cleanup
/priority important-longterm

/assign @stevekuznetsov

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 1]
```diff
@@ -253,296 +247,41 @@ func TestListContinuationWithFilter(t *testing.T) {
...
+			t.Fatal(err)
 		}
-	}
-}
-
-func withClientConfig(config *embed.Config) setupOption {
-	return func(options *setupOptions) {
-		options.client = func(t *testing.T) *clientv3.Client {
-			return testserver.RunEtcd(t, config)
+		if _, err := etcdClient.KV.Compact(ctx, int64(rv), clientv3.WithCompactPhysical()); err != nil {
```
Comment (stevekuznetsov) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:257]: IIRC some integration tests somewhere end up needing compaction as well, so there might be utility in moving this whole thing to the testing package (?) for reuse.

**Review by stevekuznetsov — APPROVED** [review]

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:257]: I think it makes sense - but for the sake of avoiding making individual changes too large, I would prefer making that in further steps of the refactoring (if that's ok for you).

**Comment by wojtek-t** [comment]
@stevekuznetsov - I fixed gofmt issue - can you please re-tag it?

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 1 above)
Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:257]: Yep, that sounds good - was just raising awareness.

**Pull request merged**